runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/config/MqttKafkaRouteConfig.java

```diff
@@ -58,4 +60,29 @@ boolean matches(
     {
         return when.isEmpty() || topic != null && when.stream().anyMatch(m -> m.matches(topic));
     }
+
+    public boolean matchesTopicFilter(
+        String topicFilter)
+    {
+        return when.isEmpty() || when.stream().anyMatch(m -> matchTopicFilter(topicFilter, m.topic));
+    }
+
+    private boolean matchTopicFilter(
+        String topicFilter,
+        String topic)
+    {
+        Matcher topicFilterMatcher = asTopicFilterMatcher(topicFilter);
+        return topicFilterMatcher.reset(topic).matches();
```

Comment:
I'm not sure if I understand what you mean by "perhaps the opposite way".
We only know what we want to match against once the we receive the subscribe filter, which is either in mqtt begin, or flush, so I'm not sure if we can optimise here very easily.

Reply:
Let's take an example to see if there's any opportunity to pre-create a matcher here.

If the routes have subscribe when topic is `/some/hierarchical/topic/name`, then it would match the following topic filter patterns (i think)
 - `/(some|+)/(hierarchical|+)/(topic|+)/(name|+)`
 - `/(some|+)/(hierarchical|+)/(topic|+)/(name|+)/#`
 - `/(some|+)/(hierarchical|+)/(topic|+)/#`
 - `/(some|+)/(hierarchical|+)/#`
 - `/(some|+)/#`
 - `/#`
 - `#`

This can be combined into a single regex as it either terminates with `#` or continues to be more specific.

Then the topic-derived regex can be used to match the topic filter, which is opposite from how we normally match the topic to the topic filter.